.pylintrc

```diff
@@ -3,3 +3,6 @@
 argument-rgx=[a-z0-9_]{1,30}$
 attr-rgx=[a-z0-9_]{1,30}$
 variable-rgx=[a-z0-9_]{1,30}$
+
+disable=
+    too-many-arguments
```

Comment:
Should we be disabling this globally?

The linter rule is to complain because of the "code smell". Often (not always) having too many arguments is synonym of an issue with the way the problem was modeled.

Instead of generally disabling the rule, I think we should try the following (in order of preference imo):

1. Try to refactor the function so that it no longer receives so many arguments. Receiving too many arguments might be suggesting that the function is too complex, or something like that.
2. If (1) is not possible, then try to disable the rule for that specific function/sector of code.

Disabling the rule for *all* cases opens the door to abusing the absence of the rule.

Reply:
@sasalatart you are right, I will disable only when it's used.

The thing is that @jia200x refactored my refactor a couple days ago with a good reason: giving a bunch number of parameters in the function argument is a pythonic common practice in packages such as sklearn, numpy and scipy.

Your writing gave me also the idea to store the observed variable and the time vector in the first call of the .fit() function, which should reduce input errors and simplify syntax. I will take on board both although it may not always be possible.